gas-tank/balance-services.md

```diff
@@ -0,0 +1,540 @@
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const message = "0x1234"; // Message to sign // [!code highlight]
+const privateKey = "0x1234"; // Private key of the address // [!code highlight]
+const wallet = new ethers.Wallet(privateKey);
+const signature = await wallet.signMessage(message);
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/withdraw`, {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json"
+    "Authorization": `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    tokenAddress: "string", // [!code highlight]
+    amount: "string", // [!code highlight]
+    chain: "string", // [!code highlight]
+    recipient: "string", // [!code highlight]
+    message: "string", // [!code highlight]
+    signature: "string" // [!code highlight]
+  },
+})
+  .then((response) => {
+    // Balance withdrawn successfully
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Internal transfer
+
+This endpoint allows users to create an internal transfer task on the Gas Tank platform.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/transfer`, {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    tokenAddress: "string", // [!code highlight]
+    amount: "string", // [!code highlight]
+    chain: "string", // [!code highlight]
+    recipient: "string", // [!code highlight]
+    message: "string", // [!code highlight]
+    signature: "string", // [!code highlight]
+  },
+})
+  .then((response) => {
+    // Internal transfer task created successfully
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Consume balance
+
+This endpoint allows users to consume their balance on the Gas Tank platform.
+
+To consume the balance, you need to sign the message with the private key of the address you want to consume the balance from. The `signature` field is the signature of the message signed by the address owner.
+
+Same as the [Increase balance](#increase-balance)/[Withdraw balance](#withdraw-balance), you can sign a string of data using [ethers.js](https://docs.ethers.org/v5/getting-started/), below is an example of [how to sign a message using ethers.js](https://docs.ethers.org/v5/getting-started/#getting-started--signing) and consume the balance.
+
+::: code-group
+
+```javascript [Request]
+import { ethers } from "ethers";
+
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+const message = "0x1234"; // Message to sign // [!code highlight]
```

Comment:
Can you specify what is the messages to sign here (how to get it? via API) and how to sign it